125,127 new class core data, made segue clean

semik997 opened the pull request.

semik997 committed:
Created a new class, CoreDataManager to interact with CoreData

semik997 committed:
Made the transition regardless of the [0] cell

semik997 committed:
Made type casting through if let

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -8,13 +8,8 @@
-                    original: String?, summary: String?, imdb: String?)
-}
-
 protocol FavoriteDeletProtocol: AnyObject {
```
Опечатка.

PaszhaBiceps replied:
Используй более правильный неймниг (для ячейки и протокола).

semik997 committed:
Transfer all
the logic for saving movies in Core Data Manager
Moved common functions into separate classes
Fixed typos and renamed films

semik997 commented:
Внёс  все оговоренные правки

semik997 committed:
Moved the logic of getting data from the data into a separate function and CoreDataManager

semik997 commented:
Сделал правки, которые ранее обсуждали
